src/lecture/seven/array/list/CustomArrayList.java

```diff
@@ -0,0 +1,141 @@
+package lecture.seven.array.list;
+
+public class CustomArrayList<V> implements List<V> {
+    private static final int STARTING_CAPACITY = 10;
+    private static final int EMPTY_CAPACITY = 0;
+    private static final Object[] EMPTY_ARRAY = {};
+    private V value;
+    private Object[] array;
+    private int arraySize = array.length;
+
+    public CustomArrayList() {
+        this.array = EMPTY_ARRAY;
```

Comment:
by default, this constructor constructs an empty list with an initial capacity of ten.